pkg/cloud/services/eks/nodegroup.go

```diff
@@ -227,6 +227,10 @@ func (s *NodegroupService) createNodegroup() (*eks.Nodegroup, error) {
 	if managedPool.InstanceType != nil {
 		input.InstanceTypes = []*string{managedPool.InstanceType}
 	}
+	for _, instanceType := range managedPool.InstanceTypes {
```

Comment:
may be clarify in the API docs as well that setting both InstanceType and InstanceTypes will append the latter to the former? that UX might come us a surprise.

Reply:
That's already the case through validation. The code looks this way but the webhooks would prevent both of these from being set.

Reply:
https://github.com/kubernetes-sigs/cluster-api-provider-aws/blob/a16dd53553cfc688f2ce29e7ab08181572181d03/exp/api/v1beta2/awsmanagedmachinepool_webhook.go#L231-L237